relay-server/src/actors/spooler/mod.rs

```diff
@@ -670,12 +679,28 @@ impl BufferService {
         let db = SqlitePoolOptions::new()
             .connect_with(options)
             .await
-            .map_err(BufferError::SetupFailed)?;
+            .map_err(BufferError::SqlxSetupFailed)?;
 
         sqlx::migrate!("../migrations").run(&db).await?;
         Ok(())
     }
 
+    /// Creates the directories for the spool file.
+    async fn create_spool_directory(path: &Path) -> Result<(), BufferError> {
+        let Some(parent) = path.parent() else {
+            return Ok(());
+        };
+        if !parent.exists() {
```

Comment:
it can happen, that the path is just a file `spool.db`, which means that this parent will be `Some("")` (empty path) (this also would happen with `.pop` you used in previous version). Meaning that the file is relative to the binary root, which we do not have to create. In the suggestion I also put the check if the path is just an empty string, just skip the creation.